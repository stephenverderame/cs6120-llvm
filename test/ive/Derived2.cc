#include <cstdio>
int main()
{
    for (int I = 0; I < 10; ++I) {
        int J = I * 10;
        printf("%d\n", J);
    }

    for (int I = 10; I >= 0; --I) {
        int K = I * 20 + 4;
        printf("%d\n", K);
    }
}