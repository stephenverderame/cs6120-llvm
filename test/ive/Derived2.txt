0
10
20
30
40
50
60
70
80
90
204
184
164
144
124
104
84
64
44
24
4
