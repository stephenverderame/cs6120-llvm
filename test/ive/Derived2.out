Loop: 
10:                                               ; preds = %17, %9
  %.0 = phi i32 [ 10, %9 ], [ %18, %17 ]
  %11 = icmp sge i32 %.0, 0
  br i1 %11, label %13, label %12

Basic IV:   %.0 = phi i32 [ 10, %9 ], [ %18, %17 ]
Derived IV: %14 = %.0 * 20
Derived IV: %15 = %.0 * 20 + 4
Basic IV:   %18 = add nsw i32 %.0, -1
Loop: 
1:                                                ; preds = %7, %0
  %.01 = phi i32 [ 0, %0 ], [ %8, %7 ]
  %2 = icmp slt i32 %.01, 10
  br i1 %2, label %4, label %3

Basic IV:   %.01 = phi i32 [ 0, %0 ], [ %8, %7 ]
Derived IV: %5 = %.01 * 10
Basic IV:   %8 = add nsw i32 %.01, 1
